nixos/modules/virtualisation/hcloud-config.nix

```diff
@@ -0,0 +1,131 @@
+# Configuration for Hetzner Cloud virtual servers.
+
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.hcloud;
+  dynamicHostname = config.networking.hostName == "";
+in
+
+{
+  imports = [
+    ../profiles/qemu-guest.nix
+  ];
+
+  options = {
+    hcloud = {
+      efi = lib.mkOption {
+        default = pkgs.stdenv.hostPlatform.isAarch64;
+        defaultText = lib.literalExpression "pkgs.stdenv.hostPlatform.isAarch64";
+        internal = true;
+        description = ''
+          Whether the server is using EFI to boot.
+        '';
+      };
+      fetchMetadata = lib.mkOption {
+        default = false;
+        description = ''
+          Whether to make server metadata available as
+          {file}`/run/hcloud-metadata` and {file}`/run/hcloud-userdata`.
+        '';
+      };
+    };
+  };
+
+  config = {
+
+    boot.growPartition = true;
+
+    fileSystems."/" = {
+      device = "/dev/disk/by-label/nixos";
+      fsType = "ext4";
+      autoResize = true;
+    };
+
+    fileSystems."/boot" = lib.mkIf cfg.efi {
+      device = "/dev/disk/by-label/ESP";
+      fsType = "vfat";
+      options = [ "umask=0077" ];
+    };
+
+    boot.loader = {
+      timeout = 1;
+      grub.enable = !cfg.efi;
+      grub.device = "/dev/sda";
+      systemd-boot.enable = cfg.efi;
+      efi.canTouchEfiVariables = true;
+    };
+
+    # Set the hostname from server metadata by default.
+    networking.hostName = lib.mkDefault "";
+
+    # Allow root logins only using the SSH key that the user specified
+    # at server creation time.
+    services.openssh = {
+      enable = true;
+      settings.PermitRootLogin = "prohibit-password";
+    };
+
+    # Per: https://community.hetzner.com/tutorials/install-and-configure-ntp
+    networking.timeServers = [
+      "ntp1.hetzner.de"
+      "ntp2.hetzner.com"
+      "ntp3.hetzner.net"
+    ];
+
+    # Generate network configuration for systemd-networkd.
+    networking.useNetworkd = true;
+    networking.useDHCP = false;
+    systemd.services.systemd-network-generator-hcloud = {
+      wantedBy = [ "systemd-networkd.service" ];
+      # Per: https://github.com/stephank/systemd-network-generator-hcloud/blob/main/systemd-network-generator-hcloud.service.in
```

Comment:
Why are we not using that service file and only add a dropIn here?

Reply:
I wasn't aware this was possible. I hope I did it as intended.

Notably, I still had to set `systemd.targets.sysinit.wants`, despite the `WantedBy` in the service file. I'm not sure why it doesn't pick that up.